### PR TITLE
WIP: use generated clientset for deployment controllers and add expansion for instantiate

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

**Comment by mfojtik** [comment]
@deads2k @smarterclayton PTAL

I tested this and it seems like we can add missing methods via expansion. (i need to plumb tests && stuff but want to get agreement that expansions is the right way to do this... )

File: pkg/deploy/generated/internalclientset/typed/apps/internalversion/generated_expansion.go [hunk 1]
```diff
@@ -1,3 +1 @@
 package internalversion
```
Comment (enj) [inline pkg/deploy/generated/internalclientset/typed/apps/internalversion/generated_expansion.go]: Delete this file?

Comment (mfojtik) [reply pkg/deploy/generated/internalclientset/typed/apps/internalversion/generated_expansion.go]: no, clientgen will recreate it :-) without interface...

**Comment by mfojtik** [comment]
closing this in favor of https://github.com/openshift/origin/pull/16019

**Pull request closed without merging**